Sources/FoundationEssentials/FileManager/FileManager+Directories.swift

```diff
@@ -466,13 +466,20 @@ extension _FileManagerImpl {
     
     var currentDirectoryPath: String? {
 #if os(Windows)
-        let dwLength: DWORD = GetCurrentDirectoryW(0, nil)
-        return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength)) {
-            if GetCurrentDirectoryW(dwLength, $0.baseAddress) == dwLength - 1 {
-                return String(decodingCString: $0.baseAddress!, as: UTF16.self)
+        for _ in 0 ... 8 {
+            let dwLength: DWORD = GetCurrentDirectoryW(0, nil)
+            guard dwLength > 0 else { return nil }
+
+             if let szCurrentDirectory = withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength), {
+                if GetCurrentDirectoryW(dwLength, $0.baseAddress) == dwLength - 1 {
```

Comment:
Would it be possible to re-use the return value of this call to `GetCurrentDirectoryW` for the next iteration instead of doing another `GetDirectoryW(0, nil)` call?